core/capabilities/gateway_connector/service_wrapper.go

```diff
@@ -0,0 +1,151 @@
+	return &workflowConnectorSigner{
+		nodeAddress: (*config).NodeAddress(),
+		signerKey:   signerKey,
+		lggr:        lggr.Named("WorkflowConnectorSigner"),
+	}, nil
+}
+
+func (h *workflowConnectorSigner) Sign(data ...[]byte) ([]byte, error) {
+	return gwCommon.SignData(h.signerKey, data...)
+}
+
+func translateConfigs(f config.GatewayConnector) connector.ConnectorConfig {
+	r := connector.ConnectorConfig{}
+	if f.NodeAddress != nil {
+		r.NodeAddress = f.NodeAddress()
+	}
+
+	if f.DonID != nil {
+		r.DonId = f.DonID()
+	}
+
+	if f.Gateways != nil {
+		r.Gateways = make([]connector.ConnectorGatewayConfig, len(r.Gateways))
+		for index, element := range f.Gateways() {
+			r.Gateways[index] = connector.ConnectorGatewayConfig{Id: element.ID(), URL: element.URL()}
+		}
+	}
+
+	if !reflect.ValueOf(f.WSHandshakeTimeoutMillis).IsZero() {
+		r.WsClientConfig = network.WebSocketClientConfig{HandshakeTimeoutMillis: f.WSHandshakeTimeoutMillis()}
+	}
+
+	if f.AuthMinChallengeLen != nil {
+		r.AuthMinChallengeLen = f.AuthMinChallengeLen()
+	}
+
+	if f.AuthTimestampToleranceSec != nil {
+		r.AuthTimestampToleranceSec = f.AuthTimestampToleranceSec()
+	}
+	return r
+}
+
+// NOTE: this wrapper is needed to make sure that our services are started after Keystore.
+func NewGatewayConnectorServiceWrapper(config *gwConfig.GatewayConnector, keystore keystore.Eth, lggr logger.Logger) *serviceWrapper {
+	return &serviceWrapper{
+		config:   config,
+		keystore: keystore,
+		lggr:     lggr,
+	}
+}
+
+func (e *serviceWrapper) Start(ctx context.Context) error {
+	return e.StartOnce("GatewayConnectorServiceWrapper", func() error {
+		conf := *e.config
+		e.lggr.Infow("Starting GatewayConnectorServiceWrapper", "chainID", conf.ChainIDForNodeKey())
+		chainId, _ := new(big.Int).SetString(conf.ChainIDForNodeKey(), 0)
+		enabledKeys, err := e.keystore.EnabledKeysForChain(ctx, chainId)
+		if err != nil {
+			return err
+		}
+		if len(enabledKeys) == 0 {
+			return errors.New("no available keys found")
+		}
+		configuredNodeAddress := common.HexToAddress(conf.NodeAddress())
+		idx := slices.IndexFunc(enabledKeys, func(key ethkey.KeyV2) bool { return key.Address == configuredNodeAddress })
+		if idx == -1 {
+			return errors.New("key for configured node address not found")
+		}
+		signerKey := enabledKeys[idx].ToEcdsaPrivKey()
+		if enabledKeys[idx].ID() != conf.NodeAddress() {
+			return errors.New("node address mismatch")
+		}
+
+		signer, err := NewWorkflowConnectorSigner(e.config, signerKey, e.lggr)
+		if err != nil {
+			return err
+		}
+		translated := translateConfigs(conf)
+		// cannot use signer (variable of type *workflowConnectorSigner) as connector.GatewayConnectorHandler value in argument to connector.NewGatewayConnector: *workflowConnectorSigner does not implement connector.GatewayConnectorHandler (missing method Close)compilerInvalidIfaceAssign
+		e.connector, err = connector.NewGatewayConnector(&translated, signer, signer, clockwork.NewRealClock(), e.lggr)
```

Comment:
For now, you still have to implement GatewayConnectorHandler interface, otherwise stuff won't build. So leave those empty methods there - we'll remove later.